src_c/display.c

```diff
@@ -1067,8 +1125,16 @@ pg_flip(PyObject *self)
     if (state->using_gl) {
         SDL_GL_SwapWindow(win);
     }
-    else
+    else {
+        if (pg_renderer != NULL){
+            SDL_Surface *screen = pgSurface_AsSurface(pg_GetDefaultWindowSurface());
+            SDL_UpdateTexture(pg_texture, NULL, screen->pixels, screen->pitch);
```

Comment:
Should we test whether this is faster than scaling up a surface in software, just to be sure? Streaming to a texture is pretty slow

Reply:
In an ideal world, SDL_Renderer magically uses the right way to scale things, which might even be software? I found that streaming to a GPU texture is faster than scaling up in software on PC (mesa, nvidia) and Android (nexus 7, ouya), but not on RaspberryPI, but that was when using pygame 1.9.4 and modernGL or PGS4A (pygame 1.9.1? and GLES).